### PR TITLE
Update ERC-6224: Move to Review

### PR DESCRIPTION
**Pull request by Arvolear**
Updated the specification, cleaned up texts, and added proper reference implementation.

### CODE REVIEW

File: ERCS/erc-6224.md [hunk 1]
```diff
@@ -13,241 +13,613 @@ requires: 1967, 5750
 
 ## Abstract
 
-The EIP standardizes the management of smart contracts within the decentralized application ecosystem. It enables protocols to become upgradeable and reduces their maintenance threshold. This EIP additionally introduces a smart contract dependency injection mechanism to audit dependency usage, to aid larger composite projects.
+The EIP introduces a system of smart contracts that a decentralized protocol may use to manage its on-chain dependencies. The system also provides a convenient way of upgrading registered smart contracts, together with specifying custom access control permissions required by a protocol's business logic.
```
Comment (SamWilsn) [inline ERCS/erc-6224.md:16]:
```suggestion
This EIP introduces a system of smart contracts that a decentralized protocol may use to manage its on-chain dependencies. The system also provides a convenient way of upgrading registered smart contracts, together with specifying custom access control permissions required by a protocol's business logic.
```

Comment (SamWilsn) [inline ERCS/erc-6224.md:16]: I'd like to see a bit more of a technical overview in the abstract. This section should give the reader a high-level understanding of how the proposed solution is implemented.

Comment (Arvolear) [reply ERCS/erc-6224.md:16]: Roger that.

File: ERCS/erc-6224.md [hunk 2]
```diff
@@ -13,241 +13,617 @@ requires: 1967, 5750
 
 ## Abstract
 
-The EIP standardizes the management of smart contracts within the decentralized application ecosystem. It enables protocols to become upgradeable and reduces their maintenance threshold. This EIP additionally introduces a smart contract dependency injection mechanism to audit dependency usage, to aid larger composite projects.
+This EIP introduces an on-chain registry system that a decentralized protocol may use to manage its smart contracts.
+
+The proposed system consists of two components: `ContractsRegistry` and `Dependant`. The `ContractsRegistry` contract stores reference to every smart contract used within a protocol, optionally making them upgradeable by deploying self-managed proxies on top, and acts as a hub the `Dependant` contracts query to fetch their required dependencies from.
```
Comment (SamWilsn) [inline ERCS/erc-6224.md:18]:
```suggestion
The proposed system consists of two components: `ContractsRegistry` and `Dependant`. The `ContractsRegistry` contract stores references to every smart contract used within a protocol, optionally making them upgradeable by deploying self-managed proxies on top, and acts as a hub the `Dependant` contracts query to fetch their required dependencies from.
```

File: assets/erc-6224/diagram.png
Comment (SamWilsn) [inline assets/erc-6224/diagram.png:1]: If you can use an SVG, we'd appreciate it!

**Review by SamWilsn — APPROVED** [review]
Please fix these comments when you have the chance!

**Comment by Arvolear** [comment]
Hey @SamWilsn, please take a look.

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**